test/flake.nix

```diff
@@ -23,26 +23,23 @@
         inputs.check-flake.flakeModule
       ];
       perSystem = { self', pkgs, ... }: {
```

Comment:
I don't think using the local `pkgs` is the right thing to do. It should come from the environment where it's imported, so all we can really need to do is `flake.haskellFlakeProjectModules.foo = { pkgs, ... }: { overrides = f pkgs; };`.

If a flake does need to reference packages it defines, it can do so with `getSystem` and the like.